orchestrator/careplancontributor/service.go

```diff
@@ -259,7 +244,43 @@ func (s Service) withSessionOrBearerToken(next func(response http.ResponseWriter
 	}
 }
 
-func (s Service) handleNotification(_ any) error {
-	log.Info().Msg("TODO: Handle received notification")
+func (s Service) handleNotification(resource any) error {
+	notification := resource.(*coolfhir.SubscriptionNotification)
+	if notification == nil {
+		return &coolfhir.ErrorWithCode{
+			Message:    "failed to cast resource to notification",
+			StatusCode: http.StatusInternalServerError,
+		}
+	}
+
+	focusReference, err := notification.GetFocus()
+	if err != nil {
+		return err
+	}
+	if focusReference == nil {
+		return &coolfhir.ErrorWithCode{
+			Message:    "Notification focus now found",
+			StatusCode: http.StatusInternalServerError,
+		}
+	}
+
+	switch *focusReference.Type {
+	case "Task":
+		// Get task
+		var task fhir.Task
+		err = s.carePlanServiceClient.Read(*focusReference.Reference, &task)
+		if err != nil {
+			return err
+		}
+
+		// TODO: How to differentiate between create and update? (Currently we only use Create in CPS. There is code for Update but nothing calls it)
+		err = s.handleTaskFillerCreate(&task)
```

Comment:
Shouldn't this specific CPC node verify that this Task is created for them (via the `Task.owner`)? 

I'm unsure how the notification logic works now, but I can imagine the placer's CPC is also informed when they create a Task. Could also be added inside the `handleTaskFillerCreate()` call

Reply:
Hmm, that's a good point. We do have validation for creating a task and the notification is sent based on the Task.Owner so I think we should be fine, but it might be a nice extra layer of validation. Will discuss this with Rein on Monday and see if it will be required

Reply:
any SCP-node can send any notification, meaning they're from an untrusted source. So they need to be validated.

> We do have validation for creating a task and the notification is sent based on the Task.Owner

You refer to the checks done by the CPS here, right? Since the CPS is external (or might be), we can't rely on that, thus `handleTaskFillerCreate` should validate it partakes in the `Task`. Furthermore; it should check whether it's actually the filler of the Task (because if it's the placer itself, which might/is also notified, it obviously shouldn't start responding to the Task it created itself).